File: Two-Pointer/sortColors.py
from typing import List

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        low = 0
        mid = 0
        high = n - 1

        while mid <= high:
            if nums[mid] == 2:
                self.swap(nums,mid,high)
                high -= 1
            elif nums[mid] == 1:
                mid += 1
            else :
                self.swap(nums,mid,low)
                low += 1
        
    def swap(self,nums: List[int],i : int,j : int)-> None:
        temp = nums[i]
        nums[i] = nums[j]
        nums[j] = temp
        

File: Array/sortColors.js
function sortColors(nums) {
  let zeroPos = 0;
  let onePos = 0;
  let twoPos = nums.length - 1;

  while (onePos < twoPos) {
    if (nums[onePos] === 0) {
      swap(zeroPos, onePos, nums);
      zeroPos++;
      onePos++;
    } else if (nums[onePos] === 2) {
      swap(onePos, twoPos, nums);
      twoPos--;
    } else onePos++;
  }

  return nums;
}

function swap(i, j, arr) {
  const temp = arr[i];
  arr[i] = arr[j];
  arr[j] = temp;
}
